eZ/Publish/API/Repository/Tests/FieldType/UserIntegrationTest.php

```diff
@@ -466,21 +493,36 @@ public function testRemoveFieldDefinition()
 
     public function testAddFieldDefinition()
     {
-        $this->markTestIncomplete(
-            'Currently cannot be tested since user can be properly created only through UserService'
-        );
+        // Field cannot be added to ContentType with existing content.
+        $this->expectException(BadStateException::class);
+
+        return parent::testAddFieldDefinition();
     }
 
     /**
-     * @param mixed $failingValue
-     * @param string $expectedException
-     *
-     * @dataProvider provideInvalidUpdateFieldData
+     * @depends testCreateContent
      */
-    public function testUpdateContentFails($failingValue, $expectedException)
+    public function testCopyField($content)
     {
-        $this->markTestIncomplete(
-            'Currently cannot be tested since user can be properly created only through UserService'
-        );
+        // Users cannot be copied.
+        $this->expectException(UniqueConstraintViolationException::class);
```

Comment:
I feel we are mixing layers here. Dedicated exception from `eZ\Publish\API\Repository\Exceptions\` namespace should be thrown instead of `Doctrine\DBAL\Exception\UniqueConstraintViolationException` but I guess it would be hard to achieve.

Reply:
> I feel we are mixing layers here. Dedicated exception from `eZ\Publish\API\Repository\Exceptions\` namespace should be thrown instead of `Doctrine\DBAL\Exception\UniqueConstraintViolationException` but I guess it would be hard to achieve.

In this context handling this exception when updating/inserting record can be done in the form of re-throwing API exception, done via f43ce1883